artman/tasks/gapic_tasks.py

```diff
@@ -51,6 +51,32 @@ def validate(self):
         return [gapic_requirements.ConfigGenRequirements]
 
 
+class DiscoGapicConfigGenTask(task_base.TaskBase):
+    """Generates GAPIC config file from a Discovery document"""
+    default_provides = 'gapic_config_path'
+
+    def execute(self, toolkit_path, discovery_doc,
+        output_dir, api_name, api_version, organization_name):
+        api_full_name = task_utils.api_full_name(
+            api_name, api_version, organization_name)
+        config_gen_dir = os.path.join(
+            output_dir, api_full_name + '-config-gen')
+        self.exec_command(['mkdir', '-p', config_gen_dir])
+        config_gen_path = os.path.join(config_gen_dir,
+                                       api_full_name + '_gapic.yaml')
+        args = [
+            '--discovery_doc=' + os.path.abspath(discovery_doc),
```

Comment:
You probably want an explicit `os.path.expanduser`, because `abspath` and `realpath` do not resolve `~` (sadly).

```python
os.path.abspath(os.path.expanduser(discovery_doc))
```

Reply:
Done.